nixos/modules/system/boot/resolved.nix

```diff
@@ -138,6 +138,10 @@ in
 
     users.users.resolved.group = "systemd-resolve";
 
+    # add resolve to nss hosts database if enabled and nscd enabled
+    # system.nssModules is configured in nixos/modules/system/boot/systemd.nix
```

Comment:
This isn't enough. We need to mkOrder in such a way that we're sure `dns` follows `resolve`

Reply:
`dns` is still added with `(mkAfter [ "dns" ])` - try `nix-build nixos/tests/networking.nix --arg networkd true -A dhcpOneIf.driver && result/bin/nixos-run-vms` and `cat /etc/nsswitch.conf` on the client:

```
cat /etc/nsswitch.conf | grep hosts
hosts:    files mymachines resolve [!UNAVAIL=return] dns myhostname
```

Reply:
perfect